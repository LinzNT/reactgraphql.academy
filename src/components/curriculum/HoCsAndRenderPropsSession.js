import React from 'react'
import Ul, { Li } from '../layout/Ul'
import Session from './Session'

const HoCsAndRenderPropsSession = ({ hideTitle }) => (
  <Session hideTitle={hideTitle} title="FP & Advanced React Patterns I">
    <Ul>
      <Li>
        Functional programming (FP)
        <Ul>
          <Li>Composition</Li>
        </Ul>
      </Li>
      <Li>
        Advanced React patterns to reuse functionality across components
        <Ul>
          <Li>Higher-Order Components (HOCs)</Li>
          <Li>
            Declarative composition using the Render Props (AKA function as
            children)
          </Li>
        </Ul>
      </Li>
    </Ul>
  </Session>
)

export default HoCsAndRenderPropsSession
