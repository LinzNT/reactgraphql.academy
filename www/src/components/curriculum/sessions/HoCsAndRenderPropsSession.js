import React from 'react'
import Ul, { Li } from '../../layout/Ul'
import Session from './Session'

const HoCsAndRenderPropsSession = ({ title }) => (
  <Session title={title}>
    <Ul>
      <Li>
        Functional programming (FP)
        <Ul>
          <Li>Composing React components</Li>
        </Ul>
      </Li>
      <Li>
        Advanced React patterns to reuse functionality across components
        <Ul>
          <Li>Higher-Order Components (HOCs)</Li>
          <Li>
            Declarative composition using the Render Props (AKA function as
            children)
          </Li>
        </Ul>
      </Li>
    </Ul>
  </Session>
)

export default HoCsAndRenderPropsSession
