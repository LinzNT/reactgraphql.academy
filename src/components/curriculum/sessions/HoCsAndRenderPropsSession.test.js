import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import HoCsAndRenderPropsSession from './HoCsAndRenderPropsSession'

describe('<HoCsAndRenderPropsSession />', () => {
  it('should render component', () => {
    const wrapper = shallow(<HoCsAndRenderPropsSession />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
