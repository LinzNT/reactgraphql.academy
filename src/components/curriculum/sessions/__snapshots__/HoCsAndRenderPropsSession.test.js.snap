// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HoCsAndRenderPropsSession /> should render component 1`] = `
<Session>
  <styled.ul>
    <styled.li>
      Functional programming (FP)
      <styled.ul>
        <styled.li>
          Composing React components
        </styled.li>
      </styled.ul>
    </styled.li>
    <styled.li>
      Advanced React patterns to reuse functionality across components
      <styled.ul>
        <styled.li>
          Higher-Order Components (HOCs)
        </styled.li>
        <styled.li>
          Declarative composition using the Render Props (AKA function as children)
        </styled.li>
      </styled.ul>
    </styled.li>
  </styled.ul>
</Session>
`;
